guides/plugins/apps/administration/add-custom-modules.md

```diff
@@ -0,0 +1,150 @@
+* `source` (optional): The URL to your app servers endpoint from which the module is served from. This can be omitted if you want to define a menu item that should serve as a parent menu item for other app modules.
+* `parent` (optional): The administration navigation id from the menu item that serves as the parrent menu item. If omitted your module will be listed under the "My apps" menu entry. **This field will be required in future versions as we are going to remove the My Apps menu item**
+* `position` (optional): A numeric index that sets the position of your menu entry regarding to it's sibblings.
+
+Additionally you can define `label` elements inside of your `module` element, to set up how your module will be displayed in the admin menu.
+
+{% code title="manifest.xml" %}
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<manifest xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/platform/master/src/Core/Framework/App/Manifest/Schema/manifest-1.0.xsd">
+    <meta>
+        ...
+    </meta>
+    <admin>
+        <module name="exampleModule"
+                source="https://example.com/promotion/view/promotion-module"
+                parent="sw-marketing"
+                position="50"
+        >
+            <label>Example module</label>
+            <label lang="de-DE">Beispiel Modul</label>
+        </module>
+    </admin>
+</manifest>
+
+```
+{% endcode %}
+
+For a complete reference of the structure of the manifest file take a look at the [Manifest reference](../../../../resources/references/app-reference/manifest-reference.md).
+
+
+
+If the user opens the module in the administration your app will receive a request to the URL defined in the `source` attribute of your `module` element. Your app can determine the shop that has opened the module through query parameters added to the url: 
+
+* shop-id: The unique identifier of the shop, where the app was installed
+* shop-url: The URL of the shop, this can later be used to access the Shopware API
+* timestamp: The Unix timestamp when the request was created
+* shopware-shop-signature: sha256 hmac of the rest of the query string, signed with the `shop-secret`
+
+A sample request may look like this:
+
+```text
+https://example.com//promotion/view/promotion-config?shop-id=HKTOOpH9nUQ2&shop-url=http%3A%2F%2Fmy.shop.com&timestamp=1592406102&shopware-shop-signature=3621fffa80187f6d43ce6cb25760340ab9ba2ea2f601e6a78a002e601579f415
+```
+
+In this case the `shopware-shop-signature` parameter contains an sha256 hmac of the rest of the query string, signed again with the secret your app assigned the shop during the [registration](../app-base-guide.md#setup). The signature can be used to verify the authenticity of the request.
+
+## Leave loading state
+
+Because your module is displayed as an iframe in the administration, Shopware can not easily tell, when your module has finished loading.
```

Comment:
```suggestion
Because your module is displayed as an iframe in the administration, Shopware can not easily tell when your module has finished loading.
```